packages/mgt/src/components/mgt-get/mgt-get.ts

```diff
@@ -235,77 +305,95 @@ export class MgtGet extends MgtTemplatedComponent {
-                page.value = response.value.concat(page.value);
-                response = page;
-                if (!this.isPolling) {
-                  this.response = response;
+            if (isDeltaLink && this.response && Array.isArray(this.response.value) && Array.isArray(response.value)) {
+              response.value = this.response.value.concat(response.value);
+            }
+
+            if (!this.isPolling && !equals(this.response, response)) {
+              this.response = response;
+            }
+
+            // get more pages if there are available
+            if (response && Array.isArray(response.value) && response['@odata.nextLink']) {
+              let pageCount = 1;
+              let page = response;
+
+              while (
+                (pageCount < this.maxPages || this.maxPages <= 0 || (isDeltaLink && this.pollingRate)) &&
+                page &&
+                page['@odata.nextLink']
+              ) {
+                pageCount++;
+                const nextResource = page['@odata.nextLink'].split(this.version)[1];
+                page = await graph.client
+                  .api(nextResource)
+                  .version(this.version)
+                  .get();
+                if (page && page.value && page.value.length) {
+                  page.value = response.value.concat(page.value);
+                  response = page;
+                  if (!this.isPolling) {
+                    this.response = response;
+                  }
                 }
               }
             }
-          }
-        } else {
-          if (this.resource.indexOf('/photo/$value') === -1) {
-            throw new Error('Only /photo/$value endpoints support the image type');
-          }
+          } else {
+            if (this.resource.indexOf('/photo/$value') === -1) {
+              throw new Error('Only /photo/$value endpoints support the image type');
+            }
 
-          // Sanitizing the resource to ensure getPhotoForResource gets the right format
-          const sanitizedResource = this.resource.replace('/photo/$value', '');
-          const photoResponse = await getPhotoForResource(graph, sanitizedResource, this.scopes);
+            // Sanitizing the resource to ensure getPhotoForResource gets the right format
+            const sanitizedResource = this.resource.replace('/photo/$value', '');
+            const photoResponse = await getPhotoForResource(graph, sanitizedResource, this.scopes);
 
-          if (photoResponse) {
-            response = {
-              image: photoResponse.photo
-            };
+            if (photoResponse) {
+              response = {
+                image: photoResponse.photo
+              };
+            }
           }
         }
 
+        if (this.shouldUpdateCache() && response) {
+          cache = CacheService.getCache<CacheResponse>(cacheSchema, responsesStore);
+          cache.putValue(key, { response: JSON.stringify(response) });
+        }
+
```

Comment:
This block should be moved inside the previous block. The response should only be cached when it was fetched from the graph - otherwise an already cached response will get cached again, extending the invalidation period.

```suggestion

          if (this.shouldUpdateCache() && response) {
            cache = CacheService.getCache<CacheResponse>(cacheSchema, responsesStore);
            cache.putValue(key, { response: JSON.stringify(response) });
          }
        }     
```

Reply:
Definitely agree! Change made.